### PR TITLE
Pass context.Context down

### PR DESCRIPTION
**Pull request by 6543**
this will make it possible to cancel requests for top and trace processes ...

Guide: https://golangbyexample.com/using-context-in-golang-complete-guide/
Ref: https://pkg.go.dev/context

### CODE REVIEW

File: server/remote/remote.go [hunk 1]
```diff
@@ -14,63 +14,61 @@
 
 package remote
 
-//go:generate mockery -name Remote -output mock -case=underscore
+//go:generate mockery -name Remote -output mocks -case=underscore
```
Comment (6543) [inline server/remote/remote.go:17]: fix go generate

File: server/remote/bitbucketserver/internal/client.go [hunk 2]
```diff
@@ -233,9 +241,22 @@ func (c *Client) GetHooks(owner string, name string) (*HookSettings, error) {
 
 //TODO: make these as as general do with the action
 
+//Helper function to help create get
+func (c *Client) doGet(url string) (*http.Response, error) {
+	request, err := http.NewRequestWithContext(c.ctx, "PUT", url, nil)
```
Comment (anbraten) [inline server/remote/bitbucketserver/internal/client.go:246]:
```suggestion
	request, err := http.NewRequestWithContext(c.ctx, "GET", url, nil)
```

File: server/remote/coding/coding.go [hunk 3]
```diff
@@ -145,7 +145,7 @@ func (c *Coding) Refresh(u *model.User) (bool, error) {
 }
 
 // Teams fetches a list of team memberships from the remote system.
-func (c *Coding) Teams(u *model.User) ([]*model.Team, error) {
+func (c *Coding) Teams(ctx context.Context, u *model.User) ([]*model.Team, error) {
 	// EMPTY: not implemented in Coding OAuth API
 	return nil, nil
```
Comment (anbraten) [inline server/remote/coding/coding.go:150]:
```suggestion
	return nil, fmt.Errorf("Not implemented")
```

File: server/remote/gitlab3/gitlab_test.go [hunk 4]
```diff
@@ -49,15 +49,15 @@ func Test_Gitlab(t *testing.T) {
 		g.Describe("AllProjects", func() {
 			g.It("Should return only non-archived projects is hidden", func() {
 				gitlab.HideArchives = true
-				_projects, err := gitlab.Repos(&user)
+				_projects, err := gitlab.Repos(nil, &user)
```
Comment (anbraten) [inline server/remote/gitlab3/gitlab_test.go:52]: Why don't you use `ctx.Background()` in this file?

File: server/shared/configFetcher.go [hunk 5]
```diff
@@ -27,81 +29,90 @@ func NewConfigFetcher(remote remote.Remote, user *model.User, repo *model.Repo,
...
+	logrus.Tracef("ConfigFetch[%s]: user did not defined own config follow default procedure", cf.repo.FullName)
+	// no user defined config so try .woodpecker/*.yml -> .woodpecker.yml -> .drone.yml
+
+	// test .woodpecker/ folder
+	// if folder is not supported we will get a "Not implemented" error and continue
+	config = ".woodpecker"
+	files, err := cf.remote_.Dir(ctx, cf.user, cf.repo, cf.build, config)
+	logrus.Tracef("ConfigFetch[%s]: found %d files in '%s'", cf.repo.FullName, len(files), config)
+	files = filterPipelineFiles(files)
+	if err == nil && len(files) != 0 {
```
Comment (anbraten) [inline server/shared/configFetcher.go:86]:
```suggestion
	files = filterPipelineFiles(files)
	if err == nil && len(files) != 0 {
          logrus.Tracef("ConfigFetch[%s]: found %d files in '%s'", cf.repo.FullName, len(files), config)
```

File: server/remote/gitlab3/gitlab_test.go (hunk 4 above)
Comment (6543) [reply server/remote/gitlab3/gitlab_test.go:52]: because gitlab3 get deleted soon

File: server/remote/bitbucketserver/internal/client.go (hunk 2 above)
Comment (6543) [reply server/remote/bitbucketserver/internal/client.go:246]: Good catch missed that ... (copy/paste issue)

**Comment by 6543** [comment]
for gitlab3 see  #358 it will be removed ...
for gitlab I did not pass it down since I would have to extend the custom client just to throw that change after #358 got merged

so if  #358 lands first I'll add it here else it's the other way arround

**Review by anbraten — APPROVED** [review]

**Pull request merged**